superset/connectors/base/models.py

```diff
@@ -359,10 +359,7 @@ def handle_single_value(value: Optional[FilterValue]) -> Optional[FilterValue]:
         if is_list_target and not isinstance(values, (tuple, list)):
             values = [values]  # type: ignore
         elif not is_list_target and isinstance(values, (tuple, list)):
-            if values:
-                values = values[0]
-            else:
-                values = None
+            values = values[0] if values else None
```

Comment:
if values could be dangerous, as [],0, None would return false, it would be nice to be a bit more specific in the check.